### PR TITLE
fix(llmisvc): migrate tokenProcessorConfig and blockSize

### PR DESCRIPTION
**Pull request by pierDipi**
In llm-d-inference-scheduler v0.6.0, the precise-prefix-cache-scorer plugin changed its config schema: tokenProcessorConfig moved from being nested inside indexerConfig to a top-level field under the plugin parameters. Without migration, the scorer silently falls back to defaults (blockSize: 16) and produces zero scores, breaking prefix cache routing entirely.

Add WithMigrateTokenProcessorConfig mutation that detects tokenProcessorConfig nested inside indexerConfig and moves it to the top level of the plugin parameters. The migration skips if a top-level tokenProcessorConfig already exists to avoid overwriting explicit user configuration. Update sample docs and e2e fixtures to use the new schema.

the prefix-cache-scorer plugin renamed the blockSize field (characters) to blockSizeTokens (tokens).
If only the deprecated blockSize is present the plugin refuses to start. Add WithMigrateBlockSizeToBlockSizeTokens mutation that copies blockSize to blockSizeTokens when the new field is not already set.

### CODE REVIEW

File: pkg/controller/v1alpha2/llmisvc/controller_int_scheduler_config_test.go [hunk 1]
```diff
@@ -1267,6 +1273,134 @@ schedulingProfiles:
 		})
 	})
 
+	Context("tokenProcessorConfig migration from indexerConfig to top-level parameters", func() {
```
Comment (bartoszmajsak) [inline pkg/controller/v1alpha2/llmisvc/controller_int_scheduler_config_test.go:1276]: Maybe as follow-up we could:

- ensure we don't have overlapping test cases and slice it to a few files based on areas/contexts; this is getting big
- since mutating functions are exported maybe we could unit test them for more edge cases

File: pkg/controller/v1alpha2/llmisvc/scheduler.go [hunk 2]
```diff
@@ -821,6 +825,89 @@ func WithUdsTokenizerConfig(u *unstructured.Unstructured) error {
...
+
+	for _, plugin := range plugins {
+		pluginMap, ok := plugin.(map[string]interface{})
+		if !ok || pluginMap["type"] != precisePrefixCacheScorerPlugin {
+			continue
+		}
+
+		// Skip if tokenProcessorConfig already exists at the top level
+		if _, exists, _ := unstructured.NestedFieldNoCopy(pluginMap, "parameters", "tokenProcessorConfig"); exists {
+			continue
```
Comment (bartoszmajsak) [inline pkg/controller/v1alpha2/llmisvc/scheduler.go:852]: Worth signaling this in the logs at least?

File: pkg/controller/v1alpha2/llmisvc/scheduler.go [hunk 3]
```diff
@@ -821,6 +825,89 @@ func WithUdsTokenizerConfig(u *unstructured.Unstructured) error {
...
+		}
+
+		// Check if tokenProcessorConfig exists inside indexerConfig (old location)
+		tpc, found, err := unstructured.NestedFieldCopy(pluginMap, "parameters", "indexerConfig", "tokenProcessorConfig")
+		if err != nil || !found {
+			continue
```
Comment (bartoszmajsak) [inline pkg/controller/v1alpha2/llmisvc/scheduler.go:862]: Log error?

File: pkg/controller/v1alpha2/llmisvc/scheduler.go [hunk 4]
```diff
@@ -821,6 +825,89 @@ func WithUdsTokenizerConfig(u *unstructured.Unstructured) error {
 	return nil
 }
 
+// WithMigrateTokenProcessorConfig migrates tokenProcessorConfig from inside
+// indexerConfig to the top level of the plugin parameters for the
+// precise-prefix-cache-scorer plugin. This handles the schema change in
+// llm-d-inference-scheduler v0.6.0 where tokenProcessorConfig was promoted
+// from indexerConfig to a top-level plugin parameter.
+func WithMigrateTokenProcessorConfig(u *unstructured.Unstructured) error {
```
Comment (bartoszmajsak) [inline pkg/controller/v1alpha2/llmisvc/scheduler.go]: Maybe move them all to `scheduler_migration.go`?

File: docs/samples/llmisvc/precise-prefix-kv-cache-routing/llm-inference-service-qwen2-7b-gpu-kv-cache-routing.yaml [hunk 5]
```diff
@@ -19,12 +19,12 @@ spec:
             - type: single-profile-handler
             - type: precise-prefix-cache-scorer
               parameters:
+                tokenProcessorConfig:
+                  blockSize: 64       # Must match vLLM --block-size (default is 16)
```
Comment (bartoszmajsak) [inline docs/samples/llmisvc/precise-prefix-kv-cache-routing/llm-inference-service-qwen2-7b-gpu-kv-cache-routing.yaml:23]: Maybe a silly question, but do we have a way to ensure it / report mismatch?

**Review by bartoszmajsak — COMMENTED** [review]
I like the composable migrations!

Few minor comments for your consideration, but non-blocking really.

File: docs/samples/llmisvc/precise-prefix-kv-cache-routing/llm-inference-service-qwen2-7b-gpu-kv-cache-routing.yaml (hunk 5 above)
Comment (pierDipi) [reply docs/samples/llmisvc/precise-prefix-kv-cache-routing/llm-inference-service-qwen2-7b-gpu-kv-cache-routing.yaml:23]: you mean report mismatch with the vLLM config? Not yet

Comment (bartoszmajsak) [reply docs/samples/llmisvc/precise-prefix-kv-cache-routing/llm-inference-service-qwen2-7b-gpu-kv-cache-routing.yaml:23]: Yeah. That's what I meant - is tracked somewhere?

**Review by bartoszmajsak — APPROVED** [review]
/lgtm
/approved

**Review by israel-hdez — APPROVED** [review]
/lgtm
